lib/galaxy/config.py

```diff
@@ -921,6 +942,30 @@ def parse(string):
         return [parse(v) for v in allowed_origin_hostnames if v]
 
 
+def get_core_config_file():
+    return os.environ.get('GALAXY_CONFIG_FILE') or 'config/galaxy.yml'
+
+
+def reload_config_options(current_config):
+
```

Comment:
Can you use ``galaxy.util.load_app_properties`` to load the properties. I think that will be more correct - take into account environment variables.

Reply:
Yes, I think so. I was looking for this in the codebase.. didn't find, so wrote my own. So yes, will do.

Reply:
On second thought, I believe we don't need to take into account env vars here. We are watching the config file: when it changes we update the current (reloadable) config values. Those config values, initially, may have been loaded from the config file or env vars or command args - but that's irrelevant at this point, as they are being overridden by a runtime modification to the config file). I think "reloadable" implies "reload value when I manually change the config file, regardless of how that value was initially set".

Reply:
On the other hand, `load_app_properties` contains config reading logic that is more elaborate than just `yaml.safe_load`. But then there's also the overriding part from os.environ. So, that function reads from a file AND overrides values if env vars are set. I think I'll modify load_app_properties so we have the option to use it for reading only.